embetter/text/_word2vec.py

```diff
@@ -0,0 +1,139 @@
+        lowercase: Specifies whether the text should be lowercased during tokenization.
+
+    Currently the following models are supported by default:
+     - `conceptnet-numberbatch-17-06-300`
+     - `word2vec-ruscorpora-300`
+     - `word2vec-google-news-300`
+     - `glove-wiki-gigaword-50`
+     - `glove-wiki-gigaword-100`
+     - `glove-wiki-gigaword-200`
+     - `glove-wiki-gigaword-300`
+     - `glove-twitter-25`
+     - `glove-twitter-50`
+     - `glove-twitter-100`
+     - `glove-twitter-200`
+
+    **Usage**
+
+    ```python
+    import pandas as pd
+    from sklearn.pipeline import make_pipeline
+    from sklearn.linear_model import LogisticRegression
+
+    from embetter.grab import ColumnGrabber
+    from embetter.text import Word2VecEncoder
+
+    # Let's suppose this is the input dataframe
+    dataf = pd.DataFrame({
+        "text": ["positive sentiment", "super negative"],
+        "label_col": ["pos", "neg"]
+    })
+
+    # This pipeline grabs the `text` column from a dataframe
+    # which is then passed to a Word2Vec model.
+    text_emb_pipeline = make_pipeline(
+        ColumnGrabber("text"),
+        Word2VecEncoder("glove-wiki-gigaword-50")
+    )
+    X = text_emb_pipeline.fit_transform(dataf, dataf['label_col'])
+
+    # This pipeline can also be trained to make predictions, using
+    # the embedded features.
+    text_clf_pipeline = make_pipeline(
+        text_emb_pipeline,
+        LogisticRegression()
+    )
+
+    # Prediction example
+    text_clf_pipeline.fit(dataf, dataf['label_col']).predict(dataf)
+    ```
+    """
+
+    def __init__(
+        self,
+        model: Union[str, Word2Vec, KeyedVectors] = "word2vec-google-news-300",
+        agg: Literal["mean", "max", "both"] = "mean",
+        deacc: bool = False,
+        lowercase: bool = False,
+    ):
+        self.model = model
+        if isinstance(model, str):
+            if model not in downloader.info()["models"]:
+                raise ValueError("Model with the specified name not found.")
+            self.keyed_vectors: KeyedVectors = downloader.load(model)  # type: ignore
+        elif isinstance(model, Word2Vec):
+            self.keyed_vectors: KeyedVectors = model.wv
+        elif isinstance(model, KeyedVectors):
+            self.keyed_vectors: KeyedVectors = model
+        else:
+            raise TypeError(
+                f"You should pass a model name, keyed vectors or a Word2Vec model to Word2VecEncoder, not {type(model)}"
+            )
+        self.agg = agg
+        self.deacc = deacc
+        self.lowercase = lowercase
+
+    def fit(self, X, y=None):
+        """No-op. Merely checks for object inputs per sklearn standard."""
+        # Scikit-learn also expects this in the `.fit()` command.
+        self._check_inputs(X)
+        self.n_features_out = (
+            self.keyed_vectors.vector_size
+            if self.agg != "both"
+            else self.keyed_vectors.vector_size * 2
+        )
```

Comment:
Technically the `.fit()` is no longer a `no-op` because you're setting `self.n_features_out` here. Since some user assume that these methods are stateless they may call `.transform()` without calling `.fit()`, so could we move this to the `.transform()`-method?